internal/testutil/test_hcls.go

```diff
@@ -0,0 +1,142 @@
+package testutil
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+// ########################################
+// ###### RESOURCES CONSTs
+// ########################################
+
+const pfPort = "packetfabric_port"
+const pfCloudRouter = "packetfabric_cloud_router"
+const pfCloudRouterConnAws = "packetfabric_cloud_router_connection_aws"
+const pfCloudRouterBgpSession = "packetfabric_cloud_router_bgp_session"
+const pfCsAwsHostedConn = "packetfabric_cs_aws_hosted_connection"
+
+// ########################################
+// ###### HCLs FOR REQUIRED FIELDS
+// ########################################
+
+// packetfabric_port
+func HclPort() (hcl, resourceName, resourceReferece string) {
+	resourceReferece, resourceName = _generateResourceName(pfPort)
+	pop, _, media, availableSpeed, err := GetPopAndZoneWithAvailablePort(os.Getenv(PF_CRC_SPEED_KEY))
+	if err != nil {
+		log.Panic(err)
+	}
+	var portEnabled bool
+	if os.Getenv(PF_PORT_ENABLED_KEY) != "" {
+		portEnabled, err = strconv.ParseBool(os.Getenv(PF_PORT_ENABLED_KEY))
+		if err != nil {
+			log.Fatal("port enabled must be either true or false")
+		}
+	}
```

Comment:
How should the test for disabling the port look with this code? How to test in one run disabled and enabled ports? Now by default, all ports will be disabled or if env is configured, they will be enabled.

Reply:
So, the `DISABLE` and `DELETE` tests will be triggered by an `UPDATE` step, the plan is to have a separated test where it will create a new port (as disabled or enabled, based on the provided envar), then update the port state by inverting the value provided through envar.

Sample flow:

1. Create port (disabled or enabled, based on `PF_PORT_ENABLED` envar)
2. Update port with the opposite value provided by `PF_PORT_ENABLED` envar
3. Destroy/Delete port

The flow above will come on future PRs once we start implementing the exceptional/secondary test scenarios.

Reply:
Imagine I have one test intending to have a disabled port and a second with an enabled port. As the same env variable controls it, it will be hard to run them simultaneously (e.g. `go test ./...`). Also, imagine writinga test that will start with an enabled port but end with a disabled one - I will have to change process env during runtime (let's try to make it even messier - imagine running tests in parallel).

Reply:
We could use the following guidelines to determine if the variable should be put in env, as a hcl function argument, or hardcode in the test:

* env variable
  * value is confidential (e.g. API key, account ID etc.)
  * we want to be able to change the value for ALL tests in one run
* hcl template function argument  
  * The resource related to the variable is a base resource for other resources (i.e. port, cloud router) - it may change per test case
  * The variable value is highly dynamic, and it may be changed even in one test case
  * The variable value may be different in different test cases
* hardcoded
  * value is static, and we do not really care about it (as long as it is correct) as we need it only to compare state and terraform configuration, for example all IP ranges, ASN, VLANs)

Reply:
The main reason for using this envar for ports is due to a limitation on PF's dev environment where a port cannot be deleted before disabling it, @rjouhann and I had a chat about it on our internal slack. So, to address that concern of running tests in parallel, we can simple add two parameters, the first one (`disabled`) to determine the port state needed by tests that depends on `HclPort` and  a second one (`skipEnvar`) to determine if the value provided on (`disabled`) has priority over `envar`  `func HclPort(disabled, skipEnvar bool) ...`.

How does that sound?